tools/rosmsg/src/rosmsg/__init__.py

```diff
@@ -693,9 +701,16 @@ def rosmsg_cmd_list(mode, full, argv=None):
         raise ValueError('Unknown mode for iterate_packages: %s'%mode)
     rospack = rospkg.RosPack()
     packs = sorted([x for x in iterate_packages(rospack, mode)])
+    msgs_to_show = []
     for (p, direc) in packs:
         for file in _list_types(direc, subdir, mode):
-            print( "%s/%s"%(p, file))
+            if (package_name is not None) and (p != package_name):
+                continue
+            msgs_to_show.append('%s/%s'%(p, file))
+    if not msgs_to_show:
+        sys.stderr.write('WARNING: no %s is found for package [%s]\n'%(mode[1:], package_name))
```

Comment:
If no messages are found in an existing package I don't think an error message is necessary. Just outputting zero lines is ok and the return code should still be 0 since the command did the requested job successfully.